### PR TITLE
Add load outcomes script 

### PR DESCRIPTION
**Pull request by BonnieJRobert**
The source for student outcomes data has changed. We are now pulling SO data from csv's (provisioned to the project's LAN folder). 

Please review the new file `load-outcomes-data.R` with code to load new data from LAN into decimal. 
In addition, old code for connecting to the SO database was removed so please run those files and check they work.
Finally, some minor tweaks were made directing code to the new data source.

** Note to whomever does the merge => There are 2 minor merge conflicts, please keep the lines reflecting changes made in the `add-load-outcomes-script` branch.

### CODE REVIEW

**Review by jonjunduan — COMMENTED** [review]
It looks great. I can run all the R code after I added some 'overwrite = T' in dbWriteTable function. So problem is that I need to clean my schema before I run these statements. 
Only one table needs to be changed the name to match the table name we use later in other R files.

File: R/load-outcomes-data.R [hunk 1]
```diff
@@ -31,53 +31,93 @@
...
+# read csv's into objects in memory.
+fls_all <- list.files(so_lan_path, pattern = ".csv", full.names = TRUE)
+tmp_table_age_fls <- list.files(so_lan_path, pattern = "qry_make_tmp_table_Age_step1_20[0-9][0-9].csv", full.names = TRUE)
+
+# read all files into current environment, processing tmp_table_Age_step1_20xx data sets separately.
+fls <- fls_all[!fls_all %in% tmp_table_age_fls] 
+fls %>%  
+  set_names(tools::file_path_sans_ext(basename(fls))) %>% 
+  map(read_csv, show_col_types = FALSE) %>%
+  imap(~ assign(..2, ..1, envir = .GlobalEnv)) %>%
```
[the remark this answers is not in the document]
Comment (BonnieJRobert) [reply R/load-outcomes-data.R:68]: I see what you mean for the tmp_table_Age series.  We can bind_rows() directly from a list.

File: R/load-outcomes-data.R [hunk 2]
```diff
@@ -31,53 +31,93 @@
...
+so_data <- c(tools::file_path_sans_ext(basename(fls)),'tmp_table_Age')
+missing <- !so_data %in% ls()
+
+if(any(missing)) {
+  warning("The following SO datasets were not loaded into current environment:")
+  so_data[missing]
+}
+
+# recast datatypes
+APPSO_Data_Final$PEN <- as.character(APPSO_Data_Final$PEN)
```
[the remark this answers is not in the document]
Comment (BonnieJRobert) [reply R/load-outcomes-data.R]: changed to APPSO_Data_01_Final

File: R/load-cohort-bgs.R [hunk 3]
```diff
@@ -18,23 +18,11 @@ library(tidyverse)
 library(RODBC)
 library(config)
 library(DBI)
-library(RJDBC)
 
 # ---- Configure LAN and file paths ----
-db_config <- config::get("pdbtrn")
-jdbc_driver_config <- config::get("jdbc")
 lan <- config::get("lan")
 source("./sql/02b-pssm-cohorts/bgs-data.sql")
```
[the remark this answers is not in the document]
Comment (BonnieJRobert) [reply R/load-cohort-bgs.R]: actually that line is no longer needed - it had the sql for outcomes

File: R/load-near-completers-ttrain.R [hunk 4]
```diff
@@ -67,7 +52,7 @@ tmp_tbl_Age <-
   janitor::clean_names(case = "all_caps")
 
 # ---- Read SO data ----
-tmp_tbl_Age_AppendNewYears <- dbGetQuery(outcomes_con, qry_make_tmp_table_Age_step1) # adjust query for correct year
+tmp_tbl_Age_AppendNewYears <- dbReadTable(decimal_con, SQL(glue::glue('"{my_schema}"."qry_make_tmp_table_Age_step1_raw"'))) # adjust query for correct year
```
[the remark this answers is not in the document]
Comment (BonnieJRobert) [reply R/load-near-completers-ttrain.R:55]: Resolved in a prior conversation.

File: R/load-cohort-dacso.R [hunk 5]
```diff
@@ -106,7 +93,7 @@ dbWriteTable(decimal_con, name = SQL(glue::glue('"{my_schema}"."t_current_region
 
 # --- Read SO DACSO data and write to decimal ----
 if (regular_run == T | ptib_flag == T) {
-t_dacso_data_part_1_stepa <- dbGetQueryArrow(outcomes_con, DACSO_Q003_DACSO_DATA_Part_1_stepA)
+t_dacso_data_part_1_stepa <- dbReadTable(decimal_con, SQL(glue::glue('"{my_schema}"."DACSO_Q003_DACSO_DATA_Part_1_stepA_raw"')))
 dbWriteTableArrow(decimal_con, name = SQL(glue::glue('"{my_schema}"."infoware_c_outc_clean_short_resp"')), infoware_c_outc_clean_short_resp_dat)
```
[the remark this answers is not in the document]
Comment (BonnieJRobert) [reply R/load-cohort-dacso.R]: This is related to the flags PR so leaving as is; maybe it is supposed to be in the database?

File: R/load-cohort-dacso.R [hunk 6]
```diff
@@ -106,7 +93,7 @@ dbWriteTable(decimal_con, name = SQL(glue::glue('"{my_schema}"."t_current_region
...
 
 dbWriteTableArrow(decimal_con, name = SQL(glue::glue('"{my_schema}"."t_dacso_data_part_1_stepa"')), value = t_dacso_data_part_1_stepa)
```
[the remark this answers is not in the document]
Comment (BonnieJRobert) [reply R/load-cohort-dacso.R:101]: This is related to the flags PR so leaving as is; maybe it is supposed to be in the database?

**Comment by BonnieJRobert** [comment]
Thanks for your review, @jonjunduan.

**Pull request merged**